backend/pubsub.js

```diff
@@ -322,9 +338,50 @@ const publicMethods = {
   ) {
     const server = this
 
+    const msg = typeof message === 'string' ? message : JSON.stringify(message)
+    let shortMsg
+    // Utility function to remove `data` (i.e., the GIMessage data) from a
+    // message. We need this for push notifications, which may have a certain
+    // maximum size (usually around 4 KiB)
+    const shortenPayload = () => {
+      if (!shortMsg && (typeof message === 'object' && message.type === NOTIFICATION_TYPE.ENTRY && message.data)) {
+        delete message.data
+        shortMsg = JSON.stringify(message)
+      }
+      return shortMsg
+    }
+
     for (const client of to || server.clients) {
+      if (client.endpoint) {
+        // The max length for push notifications in many providers is 4 KiB.
+        // However, encrypting adds a slight overhead of 17 bytes at the end
+        // and 86 bytes at the start.
+        if (msg.length > (4096 - 86 - 17)) {
+          if (!shortenPayload()) {
+            console.info('Skipping too large of a payload for', client.id)
```

Comment:
This should probably be `console.warn` right? And maybe with more info about what message is being skipped..

Reply:
I don't think so. This isn't an issue with the server, and that could pollute the server logs making it hard to identify real issues. This log isn't actionable information either, and logging the message being skipped also wouldn't be actionable information. Either the message is too large, or it isn't, and there's nothing we can do about it.